libs/api/domains/file-upload/src/lib/file-upload.resolver.ts

```diff
@@ -8,9 +7,13 @@ export class FileUploadResolver {
   constructor(private fileStorageService: FileStorageService) {}
 
   @Mutation(() => PresignedPost)
-  createUploadUrl(
+  async createUploadUrl(
     @Args('filename') filename: string,
-  ): Promise<S3.PresignedPost> {
-    return this.fileStorageService.generatePresignedPost(filename)
+  ): Promise<PresignedPost> {
+    const result = await this.fileStorageService.generatePresignedPost(filename)
+    return {
+      url: result.url,
+      fields: result.fields,
+    }
```

Comment:
~Can the result be returned directly?

Reply:
Yes turns out it can --> done.

Reply:
This might leak unnecessary data, or not conform to the return type :thinking: